handlers/callback_helper.go

```diff
@@ -28,6 +43,50 @@ func redirectUser(w http.ResponseWriter, r *http.Request, redirectURI string, pa
 	http.Redirect(w, r, generatedURL, http.StatusSeeOther)
 }
 
-func produceKafkaMessage() {
-	// TODO: Produce message to payment-processed topic
+// produceKafkaMessage handles creating a producer, marshalling the payment id into the correct avro schema and sending
+// the message to the topic defined in ProducerTopic
+func produceKafkaMessage(paymentID string) error {
+	cfg, err := config.Get()
+	if err != nil {
+		log.Error(fmt.Errorf("error getting config for kafka message production: [%v]", err))
+		return err
+	}
+
+	// Get a producer
+	kafkaProducer, err := producer.New(&producer.Config{Acks: &producer.WaitForAll, BrokerAddrs: cfg.BrokerAddr})
+	if err != nil {
+		log.Error(fmt.Errorf("error creating kafka producer: [%v]", err))
+		return err
+	}
+	paymentProcessedSchema, err := schema.Get(cfg.SchemaRegistryURL, ProducerSchemaName)
+	producerSchema := &avro.Schema{
+		Definition: paymentProcessedSchema,
+	}
+
+	// Prepare a message with the avro schema
+	message, err := prepareKafkaMessage(paymentID, *producerSchema)
+
+	// Send the message
+	partition, offset, err := kafkaProducer.Send(message)
+	if err != nil {
+		log.Error(fmt.Errorf("failed to send message in partition: %d at offset %d", partition, offset))
+		return err
+	}
+	return nil
+}
+
+// prepareKafkaMessage is pulled out of produceKafkaMessage() to allow unit testing of non-kafka portion of code
+func prepareKafkaMessage(paymentID string, schema avro.Schema) (*producer.Message, error) {
+	paymentProcessedMessage := paymentProcessed{PaymentSessionID: paymentID}
+	messageBytes, err := schema.Marshal(paymentProcessedMessage)
+	if err != nil {
+		log.Error(fmt.Errorf("error marshalling payment processed message: [%v]", err))
```

Comment:
Don't log this message without context. Instead, assign this string to `err` and return it to be logged in ErrorR